core/src/main/java/org/axonframework/queryhandling/responsetypes/AbstractResponseType.java

```diff
@@ -37,6 +38,17 @@ protected AbstractResponseType(Class<?> expectedResponseType) {
         this.expectedResponseType = expectedResponseType;
     }
 
+    protected Type unwrapFuture(Type type) {
```

Comment:
It's being a little nit picky, but this function does not necessarily mean a `Future` is 'unwrapped'. It _might_ unwrap a `Future`. So something along the lines of `unwrapIfTypeFuture(Type)` or `unwrapFutureType(Type)` would be a bit more concise.

Open for debate though, it is a minor thing after all.

Reply:
Makes sense, it felt 'itchy' to me too. If we want to be completely sure that method name will not be misunderstood, I would go with `unwrapIfTypeFuture(Type)`.